view/src/storage.rs

```diff
@@ -412,6 +457,10 @@ impl Storage {
             )
             .execute(&mut tx)
             .await?;
+
+            //Broadcast note record to channel
+
+            self.scanned_notes_tx.send(note_record).unwrap();
```

Comment:
This is slightly nitpicky, but maybe we want to skip sending the notes into the channel here, wait until /after/ calling commit, and then loop over all the records we just inserted and push them into the channel.

That way, we avoid the possibility of a data race, where someone could be notified with a NoteRecord before it's been committed to the database. 

I think this is unlikely to be a problem in practice, but ...

Reply:
> unlikely to be a problem in practice

last words famous enough to go ahead & fix it now